gobblin-metrics/src/main/java/gobblin/metrics/metric/filter/MetricTypeRegexFilter.java

```diff
@@ -0,0 +1,24 @@
+package gobblin.metrics.metric.filter;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricFilter;
+
+
+public class MetricTypeRegexFilter implements MetricFilter {
+
+  private final Pattern regex;
+  private final Matcher matcher;
+
+  public MetricTypeRegexFilter(String regex) {
+    this.regex = Pattern.compile(regex);
+    this.matcher = this.regex.matcher("");
+  }
+
+  @Override
+  public boolean matches(String name, Metric metric) {
+    return this.matcher.reset(metric.getClass().getSimpleName()).matches();
```

Comment:
Is there a reason to use regex for the metric types? There are only five types of metrics, so it is easy to make it just a list. Using regex, on the other hand, requires for example matching capitalization of the class names.